packages/framer-motion/src/components/AnimatePresence/PresenceChild.tsx

```diff
@@ -34,40 +34,38 @@ export const PresenceChild = ({
     const presenceChildren = useConstant(newChildrenMap)
     const id = useId()
 
-    const memoizedOnExitComplete = useCallback(
-        (childId: string) => {
-            presenceChildren.set(childId, true)
-
-            for (const isComplete of presenceChildren.values()) {
-                if (!isComplete) return // can stop searching when any is incomplete
-            }
-
-            onExitComplete && onExitComplete()
-        },
-        [presenceChildren, onExitComplete]
-    )
-
-    const context = useMemo(
-        (): PresenceContextProps => ({
+    let isReusedContext = true
+    let context = useMemo((): PresenceContextProps => {
+        isReusedContext = false
+        return {
             id,
             initial,
             isPresent,
             custom,
-            onExitComplete: memoizedOnExitComplete,
+            onExitComplete: (childId: string) => {
+                presenceChildren.set(childId, true)
+
+                for (const isComplete of presenceChildren.values()) {
+                    if (!isComplete) return // can stop searching when any is incomplete
+                }
+
+                onExitComplete && onExitComplete()
+            },
```

Comment:
I reintegrated this callback to eliminate a hook now that the functions in the context object only get recreated when their dependencies change even when presenceAffectsLayout is true.

Additionally, the register function was not previously afforded this memoization behavior and now is which is a nice plus.

And we are using fewer hooks